src/frontend/screens/Settings/ProjectSettings/YourTeam/SelectDevice.tsx

```diff
@@ -91,4 +111,45 @@ const styles = StyleSheet.create({
     padding: 20,
     paddingTop: 40,
   },
+  deviceListContainer: {
+    gap: 10,
+  },
 });
+
+/**
+ * Applies specialized, context-specific behavior on top of `useLocalPeers()`
+ * by only including peers that were *initially* connected from the point of view of the consuming component.
+ * If these included peers receive subsequent updates (e.g. disconnecting), they are still returned.
+ * This hook only prevents *initially* returning peers that are disconnected.
+ */
+function useInitiallyConnectedPeers() {
```

Comment:
naming is a little confusing but hopefully these comments help clarify what this hook is doing. Basically it's keeping a slightly opinionated list of relevant device ids during the lifetime of the consuming component and using that to determine which of the local peers (in global state) should be returned